app/react/src/client/preview/render.tsx

```diff
@@ -68,18 +68,18 @@ const getReactRoot = async (el: Element): Promise<ReactRoot | null> => {
   if (!isUsingNewReactRootApi) {
     return null;
   }
+  // FIXME
+  // let root = nodes.get(el);
 
-  let root = nodes.get(el);
+  // if (!root) {
+  //   // eslint-disable-next-line import/no-unresolved
+  //   const reactDomClient = (await import('react-dom/client')).default;
```

Comment:
This import breaks on SSR in next as I guess webpack tries to resolve it or something? IDK:

```
error - ../../app/react/dist/esm/client/preview/render.js:133:0
Module not found: Can't resolve 'react-dom/client'
  131 |
  132 |             _context2.next = 8;
> 133 |             return import('react-dom/client');
  134 |
  135 |           case 8:
  136 |             reactDomClient = _context2.sent.default;

Import trace for requested module:
../../app/react/dist/esm/client/preview/config.js
./pages/_app.js

https://nextjs.org/docs/messages/module-not-found
```

Reply:
(Note this code path isn't actually running, the error happens even if there's a `return null` before the actual import line.

Reply:
Ok, I see this is similar to https://github.com/storybookjs/storybook/issues/17926.

cc @valentinpalkovic -- So the issue here is that this code is running *outside* of Storybook now in this example. So we cannot expect to be able to alter the webpack config. I fear this may not be resolvable outside of having a `@storybook/react18` package @shilman?

We might get away with it though given this is a new feature; we could say it only works with `react@18`?

@shilman if that's the way forward we need to upgrade the monorepo to `react@18`. Are you OK with me doing that in this (or a separate) PR?

Reply:
@valentinpalkovic Can we wrap this in a `try/catch`?

Reply:
Unfortunately we cannot wrap in `try/catch` as it is a build time issue. I worked around it via a custom webpack config in our example for now: https://github.com/storybookjs/storybook/pull/18159/commits/081c9902197fb171dbfe03e3b1d228f8ccdc0e19